packages/datadog-plugin-mysql/src/index.js

```diff
@@ -26,85 +33,32 @@ function createWrapQuery (tracer, config) {
         span.setTag('db.name', this.config.database)
       }
 
-      analyticsSampler.sample(span, config.measured)
+      analyticsSampler.sample(span, this.config.measured)
+      this.enter(span, store)
+    })
 
-      const sequence = scope.bind(query, span).apply(this, arguments)
+    this.addSub('apm:mysql:query:end', () => {
+      this.exit()
+    })
 
-      scope.bind(sequence)
+    this.addSub('apm:mysql:query:error', err => {
+      const span = storage.getStore().span
 
-      span.setTag('resource.name', sequence.sql)
-
-      if (sequence._callback) {
-        sequence._callback = wrapCallback(tracer, span, childOf, sequence._callback)
-      } else {
-        sequence.on('end', () => {
-          span.finish()
+      if (err) {
```

Comment:
This whole block including the condition and the individual tags can be replaced with `span.setTag('error', err)` which handles this automatically.

Reply:
@rochdev it's cheaper, perf-wise, to just check first.

Reply:
There is a check built into the function. It's also extremely low additional overhead that is worth the simpler code. And also there are upcoming changes to the core of the tracer that will actually make it cheaper to pass the error directly.